lib/biran/erb_config.rb

```diff
@@ -1,34 +1,64 @@
 module Biran
   class ERBConfig
-    attr_reader :output_dir, :source_dir, :name, :extension, :config, :output_name
-    attr_accessor :bindings
+    attr_reader :output_dir, :source_dir, :name, :extension, :config, :template_contents
+    attr_accessor :bindings, :output_name, :template_config_index
+
+    DEFAULT_TEMPLATE_CONFIG_INDEX = 1
 
     def initialize(config, name, extension, source, output_dir, output_name)
-      @name       = name
-      @extension  = extension
-      @config     = config
-      @source_dir = source
-      @output_dir = output_dir
-      @output_name = output_name
+      before_process_erb do
+        @name       = name
+        @extension  = extension
+        @config     = config
+        @source_dir = source
+        @output_dir = output_dir
+        @output_name = output_name
+      end
     end
 
     def save!
       File.open(File.join(output_dir, "#{output_name}#{extension}"), 'w') do |f|
-        f.print process_erb.result(build_erb_env.call)
+        f.print template_contents.result(build_erb_env.call)
       end
     end
 
     private
 
+    def before_process_erb
+      yield
+      begin
+        process_erb_ready?
+      rescue ArgumentError => e
+        puts 'Settings required to determine template name are not configured'
+        puts e
+        exit
```

Comment:
instead of this, you could re raise the error after the puts: 

`raise e`

Reply:
I thought raise would put the whole stack trace again, which I was trying to avoid. I see now I rescue from that and only keep the reason in `e`.